packages/ui/EventCacheable/EventCacheable.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import hoc from '@enact/core/hoc';
+
+const EventCacheable = hoc((config = {defaultEvent: null, type: null}, Wrapped) => {
+	const {defaultEvent, type} = config;
+
+	return class EventParamCacheableBase extends React.Component {
+		cachedEvent = defaultEvent || null
+
+		getEvent = () => (this.cachedEvent)
+
+		setEvent = (event) => {
+			this.cachedEvent = event;
+		}
+
+		eventHandler = (ev) => {
+			this.cachedEvent = ev;
+		}
+
+		render () {
+			const eventHandler = type ? {[type]: this.eventHandler} : null;
```

Comment:
Minor optimization but this code creates 4 new objects:
* This object
* clone of props: `{...this.props}`
* clone of this object: `{...eventHandler}`
* Merged props of `this.props` clone and `eventHandler` clone

An equally readable (imo) but more efficient alternative is:

```
const props = {...this.props};
if (type) props[type] = this.eventHandler;
```